aten/src/ATen/native/Itertools.cpp

```diff
@@ -0,0 +1,55 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+
+#include <vector>
+
+namespace {
+
+using namespace at;
+
+Tensor _triu_mask(int64_t n, int64_t dims, bool diagonal, const TensorOptions &options) {
+  // get a mask that has value 1 whose indices satisfies i < j < k < ...
+  // or i <= j <= k <= ... (depending on diagonal)
+  TensorOptions opt = std::move(options);
```

Comment:
Is this correct?

Reply:
Yeah that looks fishy

Reply:
@zou3519 @apaszke @fmassa I think that was a mistake of me.
`std::move(options)` has type `std::remove_reference<const TensorOptions&>::type` which is actually `const TensorOptions`. Also,

```C++
#include <iostream>
#include <utility>

using namespace std;

class A {
public:
    int a;
    A(int a):a(a) {}
    A(A &&rhs) {
        cout << "move constructor" << endl;
        a = rhs.a;
        rhs.a = 0;
    }
    A(const A &rhs) {
        cout << "copy constructor" << endl;
        a = rhs.a;
    }
};

void f(const A &a) {
    cout << "entering f" << endl;
    A b = a;
    cout << "leaving f" << endl;
}

void g(const A &a) {
    cout << "entering g" << endl;
    A b = move(a);
    cout << "leaving g" << endl;
}

int main() {
    A a = 3;
    f(a);
    g(a);
    cout << a.a;
}
```
outputs
```
entering f
copy constructor
leaving f
entering g
copy constructor
leaving g
3
```